src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskSearchControllerCftTest.java

```diff
@@ -868,6 +879,191 @@ public void should_have_consistent_completed_state() {
+            .body("tasks.work_type", everyItem(is(TASK_TYPE_WORK_TYPE_MAP.get(taskType))))
+            .body("total_records", greaterThanOrEqualTo(1));
+    }
+
+    @Test
+    public void should_search_by_multiple_work_types_and_return_tasks_for_each_work_type() {
+        //initiate first task
+        String taskType = "followUpOverdueReasonsForAppeal";
+        TestVariables taskVariables1 = common.setupTaskAndRetrieveIds(taskType);
+        String taskId1 = taskVariables1.getTaskId();
+
+        common.setupOrganisationalRoleAssignment(authenticationHeaders);
+
+        insertTaskInCftTaskDb(taskVariables1.getCaseId(), taskId1, taskType);
+
+        //initiate second task
+        taskType = "arrangeOfflinePayment";
+        TestVariables taskVariables2 = common.setupTaskAndRetrieveIds(taskType);
+        String taskId2 = taskVariables2.getTaskId();
+
+        insertTaskInCftTaskDb(taskVariables2.getCaseId(), taskId2, taskType);
+
+        //search by all work types
+        SearchTaskRequest searchTaskRequest = new SearchTaskRequest(asList(
+            new SearchParameter(WORK_TYPE, SearchOperator.IN,
+                TASK_TYPE_WORK_TYPE_MAP.values().stream().collect(Collectors.toList())),
+            new SearchParameter(CASE_ID, SearchOperator.IN,
+                asList(taskVariables1.getCaseId(), taskVariables2.getCaseId()))
+        ));
+
+        Response result = restApiActions.post(
+            ENDPOINT_BEING_TESTED,
+            searchTaskRequest,
+            authenticationHeaders
+        );
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.OK.value())
+            .body("tasks.size()", greaterThanOrEqualTo(2))
+            .body("tasks.jurisdiction", everyItem(is("IA")))
+            .body("tasks.case_id", contains(taskVariables2.getCaseId(), taskVariables1.getCaseId()))
+            .body("tasks.id", contains(taskId2, taskId1))
+            .body("tasks.work_type", hasItems(
+                TASK_TYPE_WORK_TYPE_MAP.get("followUpOverdueReasonsForAppeal"),
+                TASK_TYPE_WORK_TYPE_MAP.get("arrangeOfflinePayment"))
+            )
+            .body("total_records", greaterThanOrEqualTo(2));
+    }
+
+    @Test
+    public void should_return_empty_list_when_search_by_work_type_not_exists() {
+        //initiate first task
+        String taskType = "followUpOverdueReasonsForAppeal";
+        TestVariables taskVariables1 = common.setupTaskAndRetrieveIds(taskType);
+        String taskId1 = taskVariables1.getTaskId();
+
+        common.setupOrganisationalRoleAssignment(authenticationHeaders);
+
+        insertTaskInCftTaskDb(taskVariables1.getCaseId(), taskId1, taskType);
+
+        //initiate second task
+        taskType = "arrangeOfflinePayment";
+        TestVariables taskVariables2 = common.setupTaskAndRetrieveIds(taskType);
+        String taskId2 = taskVariables2.getTaskId();
+
+        insertTaskInCftTaskDb(taskVariables2.getCaseId(), taskId2, taskType);
+
```

Comment:
In this scenario, we expect that an empty list is returned if the work type did yield any results.
My suggestion for this test would be to remove the creation of a secondary task. One task created should be enough in the database

Reply:
ok